bundler/lib/bundler/lazy_specification.rb

```diff
@@ -35,7 +35,7 @@ def eql?(other)
     end
 
     def hash
-      identifier.hash
+      @__hash ||= identifier.hash
```

Comment:
This is the money maker right here.

Reply:
Was following this pattern: https://github.com/rubygems/rubygems/blob/0aa81b10c8f6c20d03747400d4386514416d7244/bundler/lib/bundler/lazy_specification.rb#L116